isogram/Isogram.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+
+
+public static class Isogram
+{
+    public static bool IsIsogram(string word)
+    {
+        HashSet<char> usedLetters = new HashSet<char>();
+        foreach (char c in word.ToLower())
+        {
+            if (charIsNotLetter(c))
+            {
+                continue;
```

Comment:
This will work, but I think it's a better practice to make the conditionals handle the flow rather than to explicitly cancel the current iteration. At least, that's how I was "raised". It's a bit cleaner. So in this case it would be more like:

```
foreach (char c in word.ToLower())
{
    if (usedLetters.Contains(c))
    {
        return false;
    }

    if (!charIsNotLetter(c)) // would just make this an "is letter" instead
    {
        usedLetters.Add(c);
    }
}
```

Reply:
@abcarp Hmm.. my intention was to _explicitly_ skip non-letter characters early in the loop. The reasoning is similar to the reason for using a guard clause and that an early return keeps the happy path of the code flow left aligned.

Here is a blog post I found on the topic. https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

I'm interested in hearing more from you here, other than just "that's how I was raised".

Reply:
Nope, that's all you get! How dare you question how I was raised?!

JK. That post is a good read, and I mostly agree with that method when it comes to returns.

My remark was about continues in the middle of loops as opposed to conditionals. I think it's somewhat like spaghetti code - makes it harder to read and can be a trap if you're maintaining code with them and happen to miss one (for larger programs). I actually looked it up to see what others thought, and apparently it's somewhat controversial.

- https://softwareengineering.stackexchange.com/questions/250706/is-continue-and-break-in-loops-antipattern-bad-practice-in-java
- https://www.teamten.com/lawrence/programming/avoid-continue.html

Take from it what you will. Either way, it seems that many agree that it's most important to not have a crap ton of nesting going on.

Reply:
Well the continue is how you align left... by flipping the conditional. Otherwise the implied happy path would be "do nothing with non-letter chars".

I'd argue that an explicit continue is more clear. Letting the non-letter chars reach the bottom of the block without doing anything with them requires us to read more code just to make sure we aren't doing something else with them. A nice "do nothing with this char" at the top of the loop is helpful imo.

I know it's not a big deal either way in this case, but it is interesting to discuss the details like this.

Reply:
I think we're maybe conflating methods/returns with loops/continues a little. That said, I don't think your way is wrong, was just giving my 2 cents on continues (I'm more in the "avoid them" camp).

Word, it's fun to dissect. It's a style thing in this case since both approaches have O(n) time complexity.

Reply:
> I think we're maybe conflating methods/returns with loops/continues a little. 

I think the concept applies either way. Early return or early "continue".